### PR TITLE
Feat/moti support component config

### PR DESCRIPTION
**Pull request by byteab**
know issues
for some reason `translateX: { type: 'spring', overshootClamping: true }` doesn't work, but if I use `transform` instead it works.
@nandorojo probably an issue in moti `useMotify()`

### CODE REVIEW

**Comment by nandorojo** [comment]
Oh sorry thought this was a PR on the moti repo

**Comment by nandorojo** [comment]
But yeah I can fix this upstream

**Comment by TheEhsanSarshar** [comment]
thanks!
yea I'll improve the code, ok I'll keep it translateX then

**Comment by nandorojo** [comment]
For shorthands, what if you just did it before the `useMotify` step. This way the `transition` has a stable reference and does less work at render time.

```ts
createAnimations(animations) {
  for (const animationKey in animations) {
    const shorthands = { x: 'translateX', y: 'translateY' }

    for (const shorthand in shorthands) {
      if (animations[animationKey][shorthand]) {
        const styleName = shorthands[shorthand]
        animations[animationKey][styleName] = animations[animationKey][shorthand]
     }
    }
  }

  return { ... }
}
```

File: packages/animations-moti/src/createAnimations.tsx [hunk 1]
```diff
@@ -172,9 +168,32 @@ export function createAnimations<A extends Record<string, MotiTransition>>(
...
-        ? props.animation[0]
+      const isAnimationArray = Array.isArray(props.animation)
+      const animationKey = isAnimationArray ? props.animation[0] : props.animation
+
+      const animationObject = isAnimationArray
+        ? props.animation
+            .filter((x) => typeof x !== 'string')
+            .reduce((a, b) => ({ ...a, ...b }))
         : props.animation
+      const normalizedAnimationObject =
```
Comment (natew) [inline packages/animations-moti/src/createAnimations.tsx:179]: lets use a for loop and a single object, should be quite a bit more performance sensitive

Comment (natew) [reply packages/animations-moti/src/createAnimations.tsx:179]: same for the animationObject above probably we can do everything in one loop

Comment (TheEhsanSarshar) [reply packages/animations-moti/src/createAnimations.tsx]: thank! updated

**Comment by TheEhsanSarshar** [comment]
@nandorojo for now we only have `x, y` so I guess it's fine to do it in a render loop?

**Pull request closed without merging**